test/blocklib/core/selector.hpp

```diff
@@ -0,0 +1,158 @@
+    std::vector<PortOut<T, Async>>         outputs;
+
+    // settings
+    A<std::uint32_t, "nInputs", Visible, Doc<"variable number of inputs">, Limits<1U, 32U>>    nInputs  = 0U;
+    A<std::uint32_t, "nOutputs", Visible, Doc<"variable number of inputs">, Limits<1U, 32U>>   nOutputs = 0U;
+    A<std::vector<std::uint32_t>, "mapIn", Visible, Doc<"input port index to route from">>     mapIn; // N.B. need two vectors since pmt_t doesn't support pairs (yet!?!)
+    A<std::vector<std::uint32_t>, "mapOut", Visible, Doc<"output port index to route to">>     mapOut;
+    A<bool, "backPressure", Visible, Doc<"true: do not consume samples from un-routed ports">> backPressure = false;
+    std::map<std::uint32_t, std::vector<std::uint32_t>>                                        _internalMapping;
+    std::uint32_t                                                                              _selectedSrc = -1U;
+
+    void
+    settings_changed(const fair::graph::property_map &old_settings, const fair::graph::property_map &new_settings) {
+        if (new_settings.contains("nInputs") || new_settings.contains("nOutputs")) {
+            fmt::print("{}: configuration changed: nInputs {} -> {}, nOutputs {} -> {}\n", static_cast<void *>(this), old_settings.at("nInputs"),
+                       new_settings.contains("nInputs") ? new_settings.at("nInputs") : "same", old_settings.at("nOutputs"), new_settings.contains("nOutputs") ? new_settings.at("nOutputs") : "same");
+            inputs.resize(nInputs);
+            outputs.resize(nOutputs);
+        }
+        if (new_settings.contains("mapIn") || new_settings.contains("mapOut")) {
+            assert(mapIn.value.size() == mapOut.value.size() && "mapIn and mapOut must have the same length");
+            _internalMapping.clear();
+
+            if (mapIn.value.size() != mapOut.value.size()) {
+                throw std::invalid_argument("Input and output map need to have the same number of elements");
+            }
+
+            for (std::size_t i = 0U; i < mapOut.value.size(); ++i) {
+                _internalMapping[mapIn.value[i]].push_back(mapOut.value[i]);
+            }
+        }
+    }
+
+    using select_reader_t  = typename PortIn<std::uint32_t, Async, Optional>::ReaderType;
+    using monitor_writer_t = typename PortOut<T, Async, Optional>::WriterType;
+    using input_reader_t   = typename PortIn<T, Async>::ReaderType;
+    using output_writer_t  = typename PortOut<T, Async>::WriterType;
+
+    fair::graph::work_return_status_t
+    process_bulk(select_reader_t                      *select, //
+                 const std::vector<input_reader_t *>  &ins,
+                 monitor_writer_t                     *monOut, //
+                 const std::vector<output_writer_t *> &outs) {
+        static_assert(std::is_same_v<std::remove_cvref_t<decltype(select)>, select_reader_t *>);
```

Comment:
This seems redundant and more unit-test-ish with the explicit declarations above? Remove?